test/Altinn.Notifications.Email.IntegrationTests/Integrations/EmailSendingConsumerTests.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System.Text.Json;
+
+using Altinn.Notifications.Email.Core;
+using Altinn.Notifications.Email.Integrations.Configuration;
+using Altinn.Notifications.Email.Integrations.Consumers;
+using Altinn.Notifications.Email.IntegrationTests.Utils;
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+using Moq;
+
+using Xunit;
+
+namespace Altinn.Notifications.Email.IntegrationTests.Integrations;
+
+public class EmailSendingConsumerTests : IDisposable
+{
+    private const string TestTopic = "email-sending";
+
+    Mock<IEmailService> _emailServiceMock;
+
+    public EmailSendingConsumerTests()
+    {
+        _emailServiceMock = new Mock<IEmailService>();
+    }
+
+    [Fact]
+    public async Task ConsumeEmailTest_Successfull_deserialization_of_message_Service_called_once()
+    {
+        // Arrange
+        await KafkaUtil.CreateTopicsAsync(TestTopic);
+
+        Core.Models.Email email =
+            new(Guid.NewGuid(), "test", "body", "fromAddress", "toAddress", Core.Models.EmailContentType.Plain);
+
+        await KafkaUtil.PostMessage(TestTopic, JsonSerializer.Serialize(email));
```

Comment:
content types will be serialized as string before we post to the topic I believe. Maybe ensure this is the case for the test as well to confirm consumer will handle string deserialization of enums